browser/ui/views/profiles/brave_profile_menu_view_browsertest.cc

```diff
@@ -96,13 +97,29 @@ class BraveProfileMenuViewTest : public InProcessBrowserTest {
     const auto* title_container_view =
         menu->identity_info_container_->children()[1].get();
     EXPECT_EQ(1u, title_container_view->children().size());
-    const auto* title_view = title_container_view->children()[0].get();
-    EXPECT_EQ(GetProfileName(),
-              static_cast<const views::Label*>(title_view)->GetText());
+    if (!browser->profile()->IsGuestSession()) {
+      const auto* title_view = title_container_view->children()[0].get();
+      EXPECT_EQ(GetProfileName(browser->profile()),
+                static_cast<const views::Label*>(title_view)->GetText());
+    }
   }
 };
 
 IN_PROC_BROWSER_TEST_F(BraveProfileMenuViewTest, TestCurrentProfileView) {
-  OpenProfileMenuView();
-  CheckIdentity();
+  OpenProfileMenuView(browser());
+  CheckIdentity(browser());
+}
+
+IN_PROC_BROWSER_TEST_F(BraveProfileMenuViewTest, OpenGuestWindowProfile) {
+  // Open a Guest window.
+  EXPECT_EQ(1U, BrowserList::GetInstance()->size());
+  ui_test_utils::BrowserChangeObserver browser_creation_observer(
+      nullptr, ui_test_utils::BrowserChangeObserver::ChangeType::kAdded);
+  profiles::SwitchToGuestProfile(base::DoNothing());
+  base::RunLoop().RunUntilIdle();
```

Comment:
Presubmit check warns about using `RunUntilIdle()` if we could avoid using it.
Below browser creation waiter below is not suffcient?

Reply:
You're right, it does work on its own, good catch. And actually, I discovered that `ui_test_utils::WaitForBrowserToOpen()` wraps this functionality a little more succinctly, so using that now!